docs/user-guide/model.rst

```diff
@@ -0,0 +1,173 @@
+A version string may simply be defined as ``version: v1``, for instance.
+
+The `schema version <../dev/internals/felis.datamodel.SchemaVersion.html>`_ object has the following
+attributes:
+
+:``current``: The current version of the schema.
+:``compatible``: A list of fully compatible versions.
+:``read_compatible``: A list of versions that are read-compatible. A read compatible version is one where a database created with the older version can be read by the newer version. An example of a non-read compatible change is removing a column from a table.
+
+While Felis does not enforce any particular format for version strings, it is recommended to use a format that
+can be compared using `semantic versioning <https://semver.org/>`__.
+
+*****
+Table
+*****
+
+A `table <../dev/internals/felis.datamodel.Schema.html#felis.datamodel.Table>`__ represents a collection of
+columns along with their indexes and constraints and has these attributes:
+
+:``name``: The name of this table. This is the name that will be used to create the table in the database.
+:``@id``: A unique identifier for this table.
+:``description``: A textual description of this table.
+:``columns``: The list of columns in the table.
+:``primaryKey``: A list of column names that make up the primary key for this table.
+:``constraints``: The list of constraints for the table. The most typical type is a foreign key constraint.
+:``indexes``: The list of indexes for the table.
+
+A table may also have the following optional attributes:
+
+:``tap:table_index``: The index of this table in a TAP schema. This is used to order the tables in the schema.
+:``mysql:charset``: The `MySQL character set <https://dev.mysql.com/doc/refman/8.4/en/charset-charsets.html>`__ for the table.
+:``mysql:engine``: The `MySQL storage engine <https://dev.mysql.com/doc/refman/8.4/en/storage-engines.html>`__ for the table.
+
+******
+Column
+******
+
+A `column <../dev/internals/felis.datamodel.Schema.html#felis.datamodel.Column>`__ represents a single field
+in a table.
+A column has a name and a data type and may have additional metadata like units of measurement.
+
+:``name``: The name of this column. This is the name that will be used to create the column in the database.
+:``@id``: A unique identifier for this column.
+:``description``: A textual description of this column.
+:``datatype``: The data type of this column. Valid datatypes are defined by the `Felis type system <../dev/internals.html#module-felis.types>`__.
+:``autoincrement``: A boolean flag indicating whether this column is auto-incrementing.
+:``length``: The length of the column. This is only applicable to certain data types.
+:``nullable``: A boolean flag indicating whether this column can be null.
+:``precision``: The precision of the column. This is current only applicable to columns with the ``timestamp`` datatype.
+:``value``: The default value for this column.
+
+A column may also have the following optional properties:
```

Comment:
Please put these in the order "ivoa:unit", "fits:tunit", and "ivoa:ucd", to keep the units together.